package/src/bin/omicron-package.rs

```diff
@@ -993,24 +1050,30 @@ async fn main() -> Result<()> {
         })?;
     }
 
-    match &args.subcommand {
+    match args.subcommand {
         SubCommand::Build(BuildCommand::Target { subcommand }) => {
             do_target(&args.artifact_dir, &args.target, &subcommand).await?;
         }
         SubCommand::Build(BuildCommand::Dot) => {
             do_dot(&get_config()?).await?;
         }
         SubCommand::Build(BuildCommand::ListOutputs { intermediate }) => {
-            do_list_outputs(&get_config()?, &args.artifact_dir, *intermediate)
+            do_list_outputs(&get_config()?, &args.artifact_dir, intermediate)
                 .await?;
         }
-        SubCommand::Build(BuildCommand::Package { disable_cache }) => {
-            do_package(&get_config()?, &args.artifact_dir, *disable_cache)
-                .await?;
+        SubCommand::Build(BuildCommand::Package { disable_cache, only }) => {
+            let mut config = get_config()?;
+            config.only = only;
```

Comment:
This is a little quirky to me - if the config already contains a `only` field, why are we unconditionally overwriting it?

If we're trying to enable the behavior that "CLI args can override a stored config file", should we at least warn if `config.only` is not empty before this assignment?

Reply:
Hmm. I don't think I even realized the config could be loaded from a file!

I don't think it makes sense for `only` to be part of a configuration file, since it's something you would want to be pretty explicit about; I only threw it onto the config struct because that was more convenient than passing the `only` vec everywhere it's needed.

Reply:
/me goes to check that it actually *can* be loaded from a file - it's been a hot minute, I just saw `get_config` and assumed it was loaded from somewhere...

Okay, looks like you're right, `get_config` isn't reading the config from a file, but rather constructing it based on arguments + reading the filesystem to determine some paths.

Ignore me! I think what you're doing here is fine.